app/controllers/users_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    @user = User.new(user_params)
+    if @user.save
+      flash.notice = 'User Added Successfully'
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      flash.notice = 'User Edited Successfully'
+      redirect_to "/users/#{params[:id]}/edit"
```

Comment:
- [ ] You should redirect to the user_path(<add_user_id_here>) instead of the same edit view if the update id successfully.  Ex: `redirect_to edit_user_path(params[:id})`

Please make this change before merging the code!